packages/query/src/index.ts

```diff
@@ -731,7 +731,7 @@ const generateQueryImplementation = ({
           )
             return param.destructured;
           return param.name === 'params'
-            ? `{ ${queryParam}: pageParam, ...params }`
+            ? `{...params, ${queryParam}: pageParam || params.page }`
```

Comment:
why params.page?

Reply:
Hello, because without this page switching does not work

Reply:
Here you force everyone to pass a page params. I use infinite query a lot without the need of this

Reply:
v5 fix this no? With `initialPageParam `?

Reply:
and for v4 it should be `{...params, ${queryParam}: pageParam || params?.['${queryParam}']}` if you want it to work no?